### ISSUE TITLE
Investigation: Solution for monitoring Tracer for Errors and Asserts in a non-failing test way

### ISSUE THREAD

**Issue by jckand** [o3de/o3de#4924]
Description:  Report.results and unexpected_lines tests for Error and Assert cause tests to fail as often for issues unrelated to test as not. Errors and Asserts may not actually cause functionality to fail, but indicate real defects in the code.
Tests get marked as unreliable since they fail on error and assert beyond the immediate test focus.
We want to see error and assert fixed and acted upon, but they may not be directly related to the success of a specific test automation case.

Details: we need a system which uses Tracer() to monitor for asserts and errors in a non-failing basis which possibly tracks event signatures with a known issue list of github issues for suppression of alerting.
Report.info is of no use since that only makes a log line, it is essentially a print() where Tracer() already logs these so info is superfluous. possibly a Report.warning adds a Result.success with a rich text message showing the error(s); warning could be called for each error in the list
note that Tracer.has_errors would need to report multiple errors if there are more than one. 
This could also work as an out of channel reporting (so not pass fail, but raises some alert message).

Acceptance Criteria: 
* design a solution which accounts for multiple errors and asserts
* design reports without pass/fail impact
* design suppression solution or tracking of known issues
* (stretch) Implement rough prototype or create a new task to prototype

**Comment by Kadino** [comment]
These messages are similar to (and probably create their own) performance issues.

If we log these error messages, Codeowners files could allow a mapping from library to source to an owning SIG.

**Comment by jromnoa** [comment]
I agree with the metrics based approach proposed by @Kadino. We could have a list of the existing `Trace::Error` or `Trace::Assert` messages and also the test that triggered it lumped together on the reporting. This will allow us the option to create an e-mail or chat message using these metrics and allow devs to know which asserts and errors need fixing as well as what test it originates from.
